src/replication/adapter/aliacr/adapter.go

```diff
@@ -0,0 +1,260 @@
+package aliacr
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"path/filepath"
+	"regexp"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cr"
+	"github.com/goharbor/harbor/src/common/utils/registry/auth"
+	"github.com/goharbor/harbor/src/replication/adapter/native"
+	"github.com/goharbor/harbor/src/replication/util"
+
+	"github.com/goharbor/harbor/src/common/utils/log"
+	adp "github.com/goharbor/harbor/src/replication/adapter"
+	"github.com/goharbor/harbor/src/replication/model"
+)
+
+func init() {
+	if err := adp.RegisterFactory(model.RegistryTypeAliAcr, func(registry *model.Registry) (adp.Adapter, error) {
+		return newAdapter(registry)
+	}); err != nil {
+		log.Errorf("failed to register factory for %s: %v", model.RegistryTypeAliAcr, err)
+		return
+	}
+	log.Infof("the factory for adapter %s registered", model.RegistryTypeAliAcr)
+}
+
+// example:
+// https://registry.%s.aliyuncs.com
+// https://cr.%s.aliyuncs.com
+var regRegion = regexp.MustCompile("https://(registry|cr)\\.([\\w\\-]+)\\.aliyuncs\\.com")
+
+func getRegion(url string) (region string, err error) {
+	if url == "" {
+		return "", errors.New("empty url")
+	}
+	rs := regRegion.FindStringSubmatch(url)
+	if rs == nil {
+		return "", errors.New("Invalid Rgistry|CR service url")
+	}
+	// fmt.Println(rs)
+	return rs[2], nil
+}
+
+func newAdapter(registry *model.Registry) (*adapter, error) {
+	region, err := getRegion(registry.URL)
+	if err != nil {
+		return nil, err
+	}
+	// fix url (allow user input cr service url)
+	registry.URL = fmt.Sprintf(registryEndpointTpl, region)
+
+	credential := NewAuth(region, registry.Credential.AccessKey, registry.Credential.AccessSecret)
+	authorizer := auth.NewStandardTokenAuthorizer(&http.Client{
+		Transport: util.GetHTTPTransport(registry.Insecure),
+	}, credential)
+	nativeRegistry, err := native.NewAdapterWithCustomizedAuthorizer(registry, authorizer)
+	if err != nil {
+		return nil, err
+	}
+
+	return &adapter{
+		region:   region,
+		registry: registry,
+		domain:   fmt.Sprintf(endpointTpl, region),
+		Adapter:  nativeRegistry,
+	}, nil
+}
+
+// adapter for to aliyun docker registry
+type adapter struct {
+	*native.Adapter
+	region   string
+	domain   string
+	registry *model.Registry
+}
+
+var _ adp.Adapter = &adapter{}
+
+// Info ...
+func (a *adapter) Info() (info *model.RegistryInfo, err error) {
+	info = &model.RegistryInfo{
+		Type: model.RegistryTypeAliAcr,
+		SupportedResourceTypes: []model.ResourceType{
+			model.ResourceTypeImage,
+		},
+		SupportedResourceFilters: []*model.FilterStyle{
+			{
+				Type:  model.FilterTypeName,
+				Style: model.FilterStyleTypeText,
+			},
+			{
+				Type:  model.FilterTypeTag,
+				Style: model.FilterStyleTypeText,
+			},
+		},
+		SupportedTriggers: []model.TriggerType{
+			model.TriggerTypeManual,
+			model.TriggerTypeScheduled,
+		},
+	}
+	return
+}
+
+// PrepareForPush nothing.
+func (a *adapter) PrepareForPush([]*model.Resource) error { return nil }
```

Comment:
The functions `PrepareForPush` and `HealthCheck` are same with the native adapter, right? If they are, we should omit them